kernel/src/transaction/mod.rs

```diff
@@ -113,6 +141,106 @@ fn with_row_tracking_cols(schema: &SchemaRef) -> SchemaRef {
     Arc::new(StructType::new_unchecked(fields))
 }
 
+/// Schema for scan row data with an additional column for new deletion vector descriptors.
+/// This is an intermediate schema used during deletion vector updates before transforming to final add actions.
+static INTERMEDIATE_DV_SCHEMA: LazyLock<SchemaRef> = LazyLock::new(|| {
+    Arc::new(StructType::new_unchecked(
+        scan_row_schema()
+            .fields()
+            .cloned()
+            .chain([StructField::nullable(
+                NEW_DELETION_VECTOR_NAME.to_string(),
+                DeletionVectorDescriptor::to_schema(),
+            )]),
+    ))
+});
+
+/// Returns the intermediate schema with deletion vector column appended to scan row schema.
+fn intermediate_dv_schema() -> &'static SchemaRef {
+    &INTERMEDIATE_DV_SCHEMA
+}
+
+/// Schema for scan row data with nullable statistics fields.
+/// Used when generating remove actions to ensure statistics can be null if missing.
+#[allow(clippy::panic)]
+static NULLABLE_SCAN_ROWS_SCHEMA: LazyLock<SchemaRef> = LazyLock::new(|| {
+    NullableStatsTransform
+        .transform_struct(scan_row_schema().as_ref())
+        .unwrap_or_else(|| panic!("Failed to transform scan_row_schema"))
+        .into_owned()
+        .into()
+});
+
+/// Returns the nullable scan row schema.
+fn nullable_scan_rows_schema() -> &'static SchemaRef {
+    &NULLABLE_SCAN_ROWS_SCHEMA
+}
+
+/// Schema for restored add actions with nullable statistics fields.
+/// Used when transforming scan data back to add actions with potentially missing statistics.
+#[allow(clippy::panic)]
+static NULLABLE_RESTORED_ADD_SCHEMA: LazyLock<SchemaRef> = LazyLock::new(|| {
+    NullableStatsTransform
+        .transform_struct(restored_add_schema())
+        .unwrap_or_else(|| panic!("Failed to transform restored_add_schema"))
+        .into_owned()
+        .into()
+});
+
+/// Returns the nullable restored add action schema.
+fn nullable_restored_add_schema() -> &'static SchemaRef {
+    &NULLABLE_RESTORED_ADD_SCHEMA
+}
+
+/// Schema for restored add actions with nullable statistics fields and a dataChange boolean field.
+/// Used when generating add actions that need to indicate whether they represent logical data changes.
+#[allow(clippy::panic)]
+static NULLABLE_RESTORED_ADD_WITH_DATA_CHANGE_SCHEMA: LazyLock<SchemaRef> = LazyLock::new(|| {
+    use crate::scan::log_replay::{CLUSTERING_PROVIDER_NAME, DEFAULT_ROW_COMMIT_VERSION_NAME};
+
+    let partition_values = MapType::new(DataType::STRING, DataType::STRING, true);
```

Comment:
can this be deireved from RESTORED_ADD_WITH_DATA_CHANGE_SCHEMA instead of of listing out all fields here?